packages/devtools_app/lib/src/screens/memory/panes/chart/controller/chart_connection.dart

```diff
@@ -38,28 +40,59 @@ class ChartConnection extends DisposableController
   Timer? _pollingTimer;
   bool _connected = false;
 
-  late final isDeviceAndroid =
-      serviceConnection.serviceManager.vm?.operatingSystem == 'android';
+  /// If true, the connection to application was lost.
+  bool _disconnected = false;
```

Comment:
the fact that we are locally storing `_connected` and `_disconnected` states seems like a bit of code smell that could lead to things getting out of sync. I don't think we should be storing these locally, but instead using the service manager as the source of truth to determine whether DevTools is connected to an app

Reply:
These values are not about connection between DevTools and application, but about connection between this instance and application. Added comment to make this clear. Thanks.